src/odemis/gui/comp/overlay/cryo_feature.py

```diff
@@ -380,3 +387,420 @@ def _get_feature_position_at_view_posture(self, feature: CryoFeature) -> Dict[st
+        """
+        evt.Skip()
+
+    def on_motion(self, evt):
+        """ Process drag motion if enabled, otherwise change cursor based on target detection/mode """
+        evt.Skip()
+
+    def _update_selected_target_position(self, v_pos):
+        """
+        Update the selected target with the newly moved position
+        :param v_pos: (int, int) the coordinates in the view
+        """
+        p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
+        self._selected_target.pos.value = [p_pos[0], p_pos[1], self._selected_target.pos.value[2]]
+        self.cnvs.update_drawing()
+
+    def _detect_point_inside_target(self, v_pos):
+        """
+        Detect if a given point is over a target
+        :param v_pos: (int, int) Point in view coordinates
+        :return: (Target or None) Found target, None if not found
+        """
+        pass
+
+    def draw(self, ctx, shift=(0, 0), scale=1.0):
+        """
+        Draw all the targets, on their location, indicating their status.
+        """
+        pass
+
+
+class CryoCorrelationFmPointsOverlay(CryoCorrelationPointsOverlay):
+
+    def _detect_point_inside_target(self, v_pos):
+        """
+        Detect if a given point is over a target
+        :param v_pos: (int, int) Point in view coordinates
+        :return: (Target or None) Found target, None if not found
+        """
+
+        def in_radius(c_x, c_y, r, x, y):
+            return math.hypot(c_x - x, c_y - y) <= r
+
+        offset = self.cnvs.get_half_buffer_size()  # to convert physical target positions to pixels
+        for target in self.tab_data.main.targets.value:
+            if "FM" in target.name.value or "POI" in target.name.value:
+                coordinates = target.coordinates.value
+                fvsp = self.cnvs.phys_to_view(coordinates, offset)
+                if in_radius(fvsp[0], fvsp[1], FEATURE_DIAMETER, v_pos[0], v_pos[1]):
+                    return target
+
+    def on_left_down(self, evt):
+        """
+        Handle mouse left click down: Create/Move feature if feature tool is toggled,
+        otherwise let the canvas handle the event (for proper dragging)
+        """
+        # Capture key presses and ignore the event
+        ctrl_mode = evt.ControlDown()
+        shift_mode = evt.ShiftDown()
+        if self.active.value and ctrl_mode:
+            self._selected_tool_va.value = TOOL_FIDUCIAL
+        elif self.active.value and shift_mode:
+            self._selected_tool_va.value = TOOL_REGION_OF_INTEREST
+
+        if self.active.value:
+            self.tab_data.focussedView.value = self.cnvs.view
+            v_pos = evt.Position
+            target = self._detect_point_inside_target(v_pos)
+            p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
+            if self._mode == MODE_EDIT_POI:
+                check_existing_poi = any("POI" in target.name.value for target in self.tab_data.main.targets.value)
```

Comment:
```suggestion
                check_existing_poi = any(target.type.value == POI for target in self.tab_data.main.targets.value)
```

Can you comment why you should check any existing POI and not check any existing FIDUCIALS? Also `if target` check is not needed for POI?

Reply:
The name and type have a different structure. The type does provide the same behavior but the name provides a deeper distinction.